sqlmesh/core/config/root.py

```diff
@@ -171,16 +176,32 @@ def _normalize_and_validate_fields(cls, values: t.Dict[str, t.Any]) -> t.Dict[st
                     f"The `{plan_deprecated}` config is deprecated. Please use the `plan.{plan_deprecated}` config instead."
                 )
 
+        if values.get("physical_schema_override") and values.get("physical_schema_mapping"):
+            raise ConfigError(
+                "Only one of `physical_schema_override` and `physical_schema_mapping` can be specified"
+            )
+
         return values
 
     @model_validator(mode="after")
     @model_validator_v1_args
     def _normalize_fields_after(cls, values: t.Dict[str, t.Any]) -> t.Dict[str, t.Any]:
         dialect = values["model_defaults"].dialect
-        values["environment_catalog_mapping"] = {
-            k: normalize_identifiers(v, dialect=dialect).name
-            for k, v in values.get("environment_catalog_mapping", {}).items()
-        }
+
+        def _normalize_identifiers(key: str) -> None:
+            values[key] = {
+                k: normalize_identifiers(v, dialect=dialect).name
+                for k, v in values.get(key, {}).items()
+            }
+
+        _normalize_identifiers("environment_catalog_mapping")
+        _normalize_identifiers("physical_schema_mapping")
```

Comment:
Was it a bug that we didn't o this previously?

Reply:
I think so. I can see that the `physical_schema_override` property was quite old and perhaps pre-dated the discovery of all the normalization issues